core/src/operations/decorators/debug.rs

```diff
@@ -13,13 +13,40 @@ pub enum DebugOptions {
     StackAll,
     /// Prints out the top n items of the stack for the current context.
     StackTop(u16),
+    /// Prints out the entire contents of RAM.
+    MemAll,
+    /// Prints out contents of memory at the provided address.
+    MemAddr(u32),
+    /// Prints out the contents of memory stored in the provided interval. Interval boundaries are
+    /// both inclusive.
+    MemInterval(u32, u32),
+    /// Prints out contents of the local at the provided index for the currently executing
+    /// procedure.
+    LocalIndex(u32),
+    /// Prints out locals stored in the provided interval of the currently executing procedure.
+    /// Interval boundaries are both inclusive.
+    LocalInterval(u32, u32, bool),
+    /// Prints out the entire state of the VM (stack and RAM).
+    All(u16),
```

Comment:
I wonder if we should reduce the number of ways we record this data. Similar to the previous comment, we can probably represent single values as special cases of intervals. Also, maybe we get rid of the `All` option for now. Basically, maybe the `DebugOptions` enum could look like this:
```Rust
pub enum DebugOptions {
    StackAll,
    StackTop(u16),
    MemAll,
    MemInterval(u32, u32),
    LocalInterval(u16, u16),
}
```
We'd still have all the previous options (except for the plain `debug`), we'd just translate `debug.mem.2` to `MemInterval(2, 2)` etc.